scripts/prepare_for_registration_bulk_creation.py

```diff
@@ -0,0 +1,155 @@
+import logging
+
+from django.core.exceptions import ValidationError
+
+from framework.celery_tasks import app as celery_app
+from website.app import setup_django
+setup_django()
+
+from osf.models import OSFUser, RegistrationProvider, RegistrationBulkUpload, \
+                       BulkUploadedRegistration, RegistrationSchema
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+@celery_app.task(name='scripts.prepare_for_registration_bulk_creation')
+def prepare_for_registration_bulk_creation(parsing_output, dry_run=False):
+
+    if not parsing_output:
+        logger.error('Missing CSV input')
+        return handle_internal_error()
+        # The following are temporarily hard-coded and used for local implementation/testing only. The format of
+        # the `csv_input` has mostly been determined, except for the `csv_parsed` which needs collaboration with
+        # the parser piece. The value of `csv_parsed` is expected to be a serializable object that can be pickled
+        # and un-pickled by celery automatically.
+        # parsing_output = {
+        #     'payload_hash': '07eef7ee-fc69-4e3a-9047-8d0e0a4cb909',
+        #     'provider_id': 'osf',
+        #     'schema_id': '60f74311ed3441000141b265',
+        #     'initiator_id': 't47ka',
+        #     'registrations': [
+        #         {
+        #             'external_id': '27093472',
+        #             'csv_raw': '1997,Ford,F250,Black',
+        #             'csv_parsed': {
+        #                 'year': 1997,
+        #                 'make': 'Ford',
+        #                 'model': 'F250',
+        #                 'color': 'Black',
+        #             },
+        #         },
+        #         {
+        #             'external_id': '98709987',
+        #             'csv_raw': '2000,Volkswagen,Golf,White',  # here goes the raw string of one row
+        #             'csv_parsed': {
+        #                 'year': 2000,
+        #                 'make': 'Volkswagen',
+        #                 'model': 'Golf',
+        #                 'color': 'White',
+        #             },
+        #         },
+        #         {
+        #             'external_id': '64510752',
+        #             'csv_raw': '2011,Subaru,WRX,Blue',  # here goes the raw string of
+        #             'csv_parsed': {
+        #                 'year': 2011,
+        #                 'make': 'Subaru',
+        #                 'model': 'WRX',
+        #                 'color': 'Blue',
+        #             },
+        #         },
+        #     ],
+        # }
+
+    logger.info('Preparing OSF database for registration bulk creation ...')
+    initiator_id = parsing_output.get('initiator_id', None)
+    if not initiator_id:
```

Comment:
Since the API `put` handler has the user, there wouldn't be any case where the initiator is undefined. I can pass the initiator_id to the task and make sure it is not `None`

Reply:
Yep. This is why I called it an internal error which shouldn't happen unless we have a coding bug. Tasks are independent jobs that needs to handle varies input. It adds lots of code since this applies to others such as `provider_id` below, I have to say, but for integrity.

Reply:
We don't have to assume the caller of the task will always provide `provider_id`. What we care is whether the `provider_id` is a valid reference to an object. We can make `provider_id` have a default value of empty string, and when we try to `get()` an object with empty string as id, there will be a `NotFoundError` and we can catch and handle that error instead.

Reply:
Agreed and verified locally. The DB query throws the exception and since we already catches it, the check I used to have is redundant. I will make the change.